www/js/control/ProfileSettings.jsx

```diff
@@ -120,6 +179,38 @@ const ProfileSettings = () => {
         console.log("notification settings before and after", notificationSettings, newNotificationSettings);
         setNotificationSettings(newNotificationSettings);
     }
+
+    async function getSyncSettings() {
+        console.log("getting sync settings");
+        var newSyncSettings = {};
+        ControlSyncHelper.getSyncSettings().then(function(showConfig) {
+            newSyncSettings.show_config = showConfig;
+            setSyncSettings(newSyncSettings);
```

Comment:
Just to clarify: this sets the in-memory value of the sync settings, but does not update the native code value in the plugin, correct? Because it seems bad to update the native code value on every `get`. Similar for the other `getXXX` calls based on native code below.

Reply:
correct, the `setSyncSettings` call is to set the React state that holds the sync settings -- which is what the UI depends on